generator/src/test/java/com/scottlogic/deg/generator/inputs/profileviolation/TypeEqualityHelper.java

```diff
@@ -0,0 +1,106 @@
+package com.scottlogic.deg.generator.inputs.profileviolation;
+
+import com.scottlogic.deg.generator.Profile;
+import com.scottlogic.deg.generator.Rule;
+import com.scottlogic.deg.generator.constraints.Constraint;
+import com.scottlogic.deg.generator.constraints.atomic.NotConstraint;
+import com.scottlogic.deg.generator.constraints.grammatical.AndConstraint;
+import com.scottlogic.deg.generator.constraints.grammatical.ConditionalConstraint;
+import com.scottlogic.deg.generator.constraints.grammatical.OrConstraint;
+import org.junit.Assert;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class TypeEqualityHelper {
+    /**
+     * Asserts all profiles in two lists have constraints and all sub-constraints of the same type.
+     * @param expectedProfileList The expected profile list.
+     * @param actualProfileList The actual profile list.
+     */
+    public static void assertListProfileTypeEquality(List<Profile> expectedProfileList, List<Profile> actualProfileList) {
```

Comment:
Assert that both lists are of the same size, otherwise a) it could give a false positive if actual is longer than expected or b) throw an out-of-range exception if actual is shorter than expected

Reply:
good spot

Reply:
Ah. So technically this would already have been caught by the sameBeanAs check but as these are now in separate files this could presumably be used without sameBeanAs